ibis/expr/format.py

```diff
@@ -1,684 +1,350 @@
 from __future__ import annotations
 
-import collections
 import functools
+import itertools
 import textwrap
-import types  # noqa: TCH003
-from typing import Any, Callable, Deque, Iterable, Mapping, Tuple
+import types
+from typing import Mapping, Sequence
 
-import rich.pretty
+from public import public
 
 import ibis
 import ibis.expr.datatypes as dt
 import ibis.expr.operations as ops
-import ibis.expr.schema as sch
 import ibis.expr.types as ir
 from ibis import util
-from ibis.common import graph
 
-Aliases = Mapping[ops.TableNode, int]
-Deps = Deque[Tuple[int, ops.TableNode]]
+_infix_ops = {
+    # comparison operations
+    ops.Equals: "==",
+    ops.IdenticalTo: "===",
+    ops.NotEquals: "!=",
+    ops.Less: "<",
+    ops.LessEqual: "<=",
+    ops.Greater: ">",
+    ops.GreaterEqual: ">=",
+    # arithmetic operations
+    ops.Add: "+",
+    ops.Subtract: "-",
+    ops.Multiply: "*",
+    ops.Divide: "/",
+    ops.FloorDivide: "//",
+    ops.Modulus: "%",
+    ops.Power: "**",
+    # temporal operations
+    ops.DateAdd: "+",
+    ops.DateSub: "-",
+    ops.DateDiff: "-",
+    ops.TimeAdd: "+",
+    ops.TimeSub: "-",
+    ops.TimeDiff: "-",
+    ops.TimestampAdd: "+",
+    ops.TimestampSub: "-",
+    ops.TimestampDiff: "-",
+    ops.IntervalAdd: "+",
+    ops.IntervalSubtract: "-",
+    ops.IntervalMultiply: "*",
+    ops.IntervalFloorDivide: "//",
+    # boolean operators
+    ops.And: "&",
+    ops.Or: "|",
+    ops.Xor: "^",
+}
+
 
+def type_info(datatype) -> str:
+    """Format `datatype` for display next to a column."""
+    return f"  # {datatype}" * ibis.options.repr.show_types
 
-class Alias:
-    __slots__ = ("value",)
 
-    def __init__(self, value: int) -> None:
-        self.value = value
+def truncate(pieces: Sequence[str], limit: int, vertical: bool = True) -> list[str]:
+    if limit < 1:
+        raise ValueError("limit must be >= 1")
+    elif limit == 1:
+        return pieces[-1]
+    elif limit >= len(pieces):
+        return pieces
+
+    first_n = limit // 2
+    last_m = limit - first_n
+    first, last = pieces[:first_n], pieces[-last_m:]
+    if vertical:
+        maxlen = max(*map(len, first), *map(len, last))
+        ellipsis = util.VERTICAL_ELLIPSIS.center(maxlen)
+    else:
+        ellipsis = util.HORIZONTAL_ELLIPSIS
```

Comment:
Seems like a bunch of this function is missing test coverage.

Reply:
Adding tests.